File: webide/src/main/java/fiveguys/webide/common/dto/ResponseDto.java
package fiveguys.webide.common.dto;

import lombok.Getter;

@Getter
public class ResponseDto<T> {
    private Integer status;
    private String message;
    private T data;

    public ResponseDto(Integer status, String message, T data) {
        this.status = status;
        this.message = message;
        this.data = data;
    }

    public static <T> ResponseDto<T> success(int status, String message, T data) {
        return new ResponseDto<>(status, message, data);
    }

    public static <T> ResponseDto<T> fail(int status, String message) {
        return new ResponseDto<>(status, message, null);
    }

}

File: webide/src/main/kotlin/fiveguys/webide/common/dto/ResponseDto.kt
package fiveguys.webide.common.dto

class ResponseDto<T>(
    val status: String,
    val message: String?,
    val data: T?
) {
    companion object {
        fun <T> success(message: String, data: T?): ResponseDto<T> {
            return ResponseDto("Success", message, data)
        }

        fun <T> fail(status: String, message: String?): ResponseDto<T> {
            return ResponseDto(status, message, null)
        }
    }
}
